verification/cbmc/stubs/hdr_write_stub.c

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include <aws/cryptosdk/private/header.h>
+
+#include <proof_helpers/make_common_data_structures.h>
+
+/*
+ * Stub for aws_cryptosdk_hdr_write
+ * The original function declares an aws_byte_buf output variable where
+ * different components of the header get written, but the actual output
+ * of the function is bytes_written. Therefore, we assign a nondet. value
+ * to this variable in case of success or zero both outbuf and outlen in
+ * case of failure
+ */
```

Comment:
nit: Suggestion:

```c
/**
 * The original aws_cryptosdk_hdr_write function declares an aws_byte_buf
 * output variable where different components of the header get written,
 * however, the actual output of the function is bytes_written. Therefore,
 * we assign a nondeterministic value to bytes_written in case of success
 * or zero both outbuf and outlen in case of failure.
 */
```

Reply:
The stub looks great. How much gain in performance?

Reply:
Thanks! It was ~40min on my dedicated machine before, but only ~10min now 😄